python/pyspark/context.py

```diff
@@ -1008,60 +1009,61 @@ def setJobGroup(self, groupId, description, interruptOnCancel=False):
         ensure that the tasks are actually stopped in a timely manner, but is off by default due
         to HDFS-1208, where HDFS may respond to Thread.interrupt() by marking nodes as dead.
 
-        .. note:: Currently, setting a group ID (set to local properties) with a thread does
-            not properly work. Internally threads on PVM and JVM are not synced, and JVM thread
-            can be reused for multiple threads on PVM, which fails to isolate local properties
-            for each thread on PVM. To work around this, you can set `PYSPARK_PIN_THREAD` to
+        .. note:: Currently, setting a group ID (set to local properties) with multiple threads
+            does not properly work. Internally threads on PVM and JVM are not synced, and JVM
+            thread can be reused for multiple threads on PVM, which fails to isolate local
+            properties for each thread on PVM.
+
+            To work around this, you can set `PYSPARK_PIN_THREAD` to
             `'true'` (see SPARK-22340). However, note that it cannot inherit the local properties
             from the parent thread although it isolates each thread on PVM and JVM with its own
-            local properties. To work around this, you should manually copy and set the local
+            local properties.
+
+            To work around this, you should manually copy and set the local
             properties from the parent thread to the child thread when you create another thread.
-        """
-        warnings.warn(
-            "Currently, setting a group ID (set to local properties) with a thread does "
-            "not properly work. "
-            "\n"
-            "Internally threads on PVM and JVM are not synced, and JVM thread can be reused "
-            "for multiple threads on PVM, which fails to isolate local properties for each "
-            "thread on PVM. "
-            "\n"
-            "To work around this, you can set PYSPARK_PIN_THREAD to true (see SPARK-22340). "
-            "However, note that it cannot inherit the local properties from the parent thread "
-            "although it isolates each thread on PVM and JVM with its own local properties. "
-            "\n"
-            "To work around this, you should manually copy and set the local properties from "
-            "the parent thread to the child thread when you create another thread.",
-            UserWarning)
+            Workaround class can be written as below (unofficial way):
+
+            >>> class CustomThread(threading.Thread):
```

Comment:
Should we make `CustomThread` as an API? I am not sure yet. Ideally we should find a way to fix it internally. This is a temp workaround for now.

Reply:
If there's an official api for this, then if user use this api to create thread, then we can prevent show the warning.

Reply:
I am not yet sure ... we should still show the warning and let users use `CustomThread` ...

Reply:
Maybe we could add a simple api like:
```
sc.createThread(thread_func, *thread_func_args)
```
and return a thread object.

Reply:
It's rather a workaround currently and we should ideally fix this rather than exposing another API for that workaround. Also to match it, we should make another API in Scala side if we go for `createThread` way.